toolchain/parse/context.cpp

```diff
@@ -303,13 +303,9 @@ auto Context::DiagnoseOperatorFixity(OperatorFixity fixity) -> void {
           "whitespace is not allowed {0:after|before} this unary operator",
           BoolAsSelect);
       emitter_->Emit(*position_, UnaryOperatorHasWhitespace, prefix);
-    } else if (IsLexicallyValidInfixOperator()) {
-      // Pre/postfix operators must not satisfy the infix operator rules.
-      CARBON_DIAGNOSTIC(
-          UnaryOperatorRequiresWhitespace, Error,
-          "whitespace is required {0:before|after} this unary operator",
-          BoolAsSelect);
-      emitter_->Emit(*position_, UnaryOperatorRequiresWhitespace, prefix);
+    } else {
+      CARBON_CHECK(!IsLexicallyValidInfixOperator(),
+                   "whitespace should have been required for an infix parse");
```

Comment:
I spent some time digging into this. It's *almost* not possible to reach this diagnostic, because almost every single one of our symbolic prefix or postfix operators happens to also be valid as an infix operator -- and so if a use were lexically valid as an infix operator, it would have been parsed as one. The exception is that the `--` and `++` operators have no infix form. And even though they can only appear at the start of a statement, it is possible to reach this diagnostic for them.

Here's a testcase that produces the "whitespace is required before this unary operator" diagnostic:
```
fn F() { var n: i32 = 0; if (false) {}--n; }
```

Reply:
Thanks! I've added a parse test, with some minimization.

Note, I also added a similar test that _doesn't_ repro, but produces a very similar parse tree so probably should, but leaving it as a todo.